opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -1466,6 +1466,36 @@ func resourceOpennebulaVirtualMachineUpdate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	if d.HasChange("cpu") || d.HasChange("vcpu") || d.HasChange("memory") {
+
+		vmState, _, _ := vmInfos.State()
+		if vmState != vm.Poweroff {
```

Comment:
This probably works but doesn't cover some other cases:

It's also possible to resize a VM in `UNDEPLOYED` state.
For ONE 6 there is even an additional note: 
> with some limitations also live in RUNNING state

[see ONE 5.12 doc](https://docs.opennebula.io/5.12/operation/vm_management/vm_instances.html#resizing-vm-capacity)
[see ONE 6 doc](https://docs.opennebula.io/6.0/management_and_operations/vm_management/vm_instances.html#resizing-vm-resources)

Did you check that Poweroff method code below won't fail if the VM is already in `POWEROFF` state ? (I didn't check for now, I'm just asking)
And if already `POWEROFF`, should we restart the VM via the `Resume` call ?

Note: for now there is no complex state management in the provider for the VM resource, it's still oversimplified, but there was a proposal in the past: #34 
Maybe that we could avoid some over complex state management to only allow the the VM resize and add a part to manage the VM state in a smarter way.

I'm just opening a discussion, feel free to discuss or just to propose something simple and good enough that could evolve later

Reply:
I think the proposal in #34 is a great feature but a heavy task. For now, I would go with a very simple check and added the UNDEPLOYED case. Is that ok for the moment?
Regarding the hotplug feature: I need to check if I can create a test setup with Opennebula 6 to test the hotplug features. Could take some time.

Reply:
I'm ok if your code only handle poweroff and undeploy states for now